openmls/src/ciphersuite/aead.rs

```diff
@@ -1,15 +1,15 @@
 use super::*;
 
 /// AEAD keys holding the plain key value and the AEAD algorithm type.
```

Comment:
You can drop the `Clone` here and for the Nonce and move it to test/test-utils only.

Reply:
It turns out that's not possible without some refactoring on the `SecretTree`. We need clone here, because we store these in the `DecryptionRatchet`, which in turn needs to implement clone, because the `SecretTree` uses `vec![None, size]` to initialize the vector of `SenderRatchet`s. I could replace that initialization, but since we're going to touch all of that anyway in #789, I'd prefer to leave a comment there indicating that the cascade of `Clone` derivations needs to be removed.

Reply:
Well, you can do something like  `(0..size.as_usize()).map(|_| None).collect()` instead of `vec!`. That's not super nice but would avoid requiring the clone until #789 gets fixed.

Reply:
Fair enough. I ended up choosing another method of initialization I found [here](https://stackoverflow.com/questions/65011997/why-cant-none-be-cloned-for-a-generic-optiont-when-t-doesnt-implement-clone), which I believe is equivalent in efficiency as `vec![...]`. Although it might all be equivalent in the end. In any case, it allowed us to finally get rid of the `Clone` derivations 🙂.